### PR TITLE
Updated PreprCMS guide

### PR DESCRIPTION
**Pull request by mlingham79**
#### Description (required)

Updated the Prepr CMS page with detailed instructions to connect an Astro project to Prepr CMS.

#### Related issues & labels (optional)

- Closes #
- Suggested label: 

 #### First-time contributor to Astro Docs

### CODE REVIEW

**Comment by sarah11918** [comment]
Hi @mlingham79! Wow, what a first contribution to Astro Docs! This is a big one, and we are so happy to receive it.  😄 

I just wanted to let you know I'll be reviewing this in the next day or so, and I've already skimmed it and it's looking great! This is one of the more polished contributions we've received from a community member by far, and it will be my pleasure to review it.

You've done an amazing job here, so I'll just be doing some structural edits to conform to our existing style, just so you know what's going on. For example, we don't use screenshots in Astro docs, so I might be adjusting some text to articulate content in words vs relying on screenshots. I'll also just be tweaking for our standard voice, verb tenses etc. Nothing major really! But, this might result in a lot of tiny "paper cuts", so please don't be overwhelmed if we end up with several small suggestions. As I said, I really do feel this is in excellent shape, thoughtful and well-written, so I just need to do some polishing so that it fits in. 

You'll see a detailed review from me soon! 💜

File: src/content/docs/en/guides/cms/preprcms.mdx [hunk 1]
```diff
@@ -6,11 +6,286 @@ service: Prepr CMS
...
+                <li>
+                    <a href={post._slug}>{post.title}</a>
+                </li>
+            ))
+        }
+    </ul>
+</Layout>
+```
+
+Now when you view the page, each article has its own link. Continue with the next step to fetch the article details.
```
Comment (sarah11918) [inline src/content/docs/en/guides/cms/preprcms.mdx:192]:
```suggestion
```
Let's combine this step with the one above. There's not much new going on here!

File: src/content/docs/en/guides/cms/preprcms.mdx [hunk 2]
```diff
@@ -6,11 +6,286 @@ service: Prepr CMS
...
+Add another query to fetch a specific article by its slug and make this page visible when a visitor clicks an article link.
+1. Create a file called `get-article-by-slug.js` in the `queries` folder and add the following query to this file to query a specific article by its slug:
```
Comment (sarah11918) [inline src/content/docs/en/guides/cms/preprcms.mdx:194]:
```suggestion
To create a page for each blog post, you will execute a new GraphQL query on a [dynamic routing]() `.astro` page. This query will fetch a specific article by its slug and a new page will be created for each individual blog post.

1. Create a file called `get-article-by-slug.js` in the `queries` folder and add the following to query a specific article by its slug and return data such as the article `title` and `content`:
```

File: src/content/docs/en/guides/cms/preprcms.mdx [hunk 3]
```diff
@@ -6,11 +6,286 @@ service: Prepr CMS
...
+         }
+       }
+     }
+   }
+}`
+
+export default GetArticleBySlug
+```
+Now that the query is added, fetch the individual article by its slug. Fetch the article title and the article content.
+
```
Comment (sarah11918) [inline src/content/docs/en/guides/cms/preprcms.mdx:221]:
```suggestion

```

File: src/content/docs/en/guides/cms/preprcms.mdx [hunk 4]
```diff
@@ -6,11 +6,286 @@ service: Prepr CMS
...
+:::tip
+You can create and [test GraphQL queries](https://docs.prepr.io/reference/graphql/v1/test-queries) using the [Apollo explorer](https://studio.apollographql.com/sandbox/explorer) in Prepr. Open the API Explorer from the *Article* content item page in Prepr.
+The Article content is stored in a *Dynamic content field*. Check out the GraphQL docs for more details on [how to fetch the data within this field](https://docs.prepr.io/reference/graphql/v1/schema-field-types-dynamic-content-field).
+:::
+
+
+2. Go to the `src/pages` folder and create a file called `[…slug].astro` file with the following code to execute the query that you just created and display the retrieved article:
```
Comment (sarah11918) [inline src/content/docs/en/guides/cms/preprcms.mdx:228]:
```suggestion
2. Inside the `src/pages` folder, create a file called `[…slug].astro`. Add the following code to import and execute the query from the previous step and display the retrieved article:
```

File: src/content/docs/en/guides/cms/preprcms.mdx [hunk 5]
```diff
@@ -6,11 +6,286 @@ service: Prepr CMS
...
+
+```astro title="src/pages/[...slug].astro"
+
```
Comment (sarah11918) [inline src/content/docs/en/guides/cms/preprcms.mdx:231]:
```suggestion
```

File: src/content/docs/en/guides/cms/preprcms.mdx [hunk 6]
```diff
@@ -6,11 +6,286 @@ service: Prepr CMS
...
+                    }
+                    {
+                        content.__typename === 'Text' &&
+                        <div set:html={content.body}></div>
+                    }
+                </div>
+            ))
+        }
+    </main>
+</Layout>
```
Comment (sarah11918) [inline src/content/docs/en/guides/cms/preprcms.mdx:261]:
```suggestion
<Layout title={article.title}>
  <main>
    <h1>{article.title}</h1>
    {
      article.content.map((content) => (
        <div>
          {
            content.__typename === "Assets" &&
            <img src={content.items[0].url} width="300" height="250"/>
          }
          {
            content.__typename === 'Text' &&
            <div set:html={content.body}></div>
          }
        </div>
      ))
    }
  </main>
</Layout>
```

File: src/content/docs/en/guides/cms/preprcms.mdx [hunk 7]
```diff
@@ -6,11 +6,286 @@ service: Prepr CMS
...
+        - get-articles.js
+        - **get-article-by-slug.js**
+- src/
+    - pages/
+      - index.astro
+      - **[…slug].astro**
+- .env
+- astro.config.mjs
+- package.json
+</FileTree>
```
Comment (sarah11918) [inline src/content/docs/en/guides/cms/preprcms.mdx:279]:
```suggestion
Your root directory should now include these new files:

<FileTree title="Project Structure">
- lib/
    - prepr.js
    - queries/
        - get-articles.js
        - **get-article-by-slug.js**
- src/
    - pages/
      - index.astro
      - **[…slug].astro**
- .env
- astro.config.mjs
- package.json
</FileTree>
```
Love that the file tree is added directly after the new file is created here!

File: src/content/docs/en/guides/cms/preprcms.mdx [hunk 8]
```diff
@@ -6,11 +6,286 @@ service: Prepr CMS
...
+
+Now, when you view your site, click an article to open the article detail page like in the image below.
```
Comment (sarah11918) [inline src/content/docs/en/guides/cms/preprcms.mdx:281]:
```suggestion
Now, when you click an article link from the main list of blog posts, you will be taken to a page with its individual content.
```

File: src/content/docs/en/guides/cms/preprcms.mdx [hunk 9]
```diff
@@ -6,11 +6,286 @@ service: Prepr CMS
...
+
+![blog article detail](https://assets-site.prepr.io//507hx87v7wkw-screenshot-2023-05-11-at-162824.png)
+
+Congratulations! You've successfully integrated an Astro project with Prepr for a simple blog.
```
Comment (sarah11918) [inline src/content/docs/en/guides/cms/preprcms.mdx:285]:
```suggestion
### Publishing your site

To deploy your Prepr blog, visit our [deployment guides](/en/guides/deploy/) and follow the instructions for your preferred hosting provider.
```

**Review by sarah11918 — COMMENTED** [review]
OK, are you ready??  😄 

I loved the structure you provided here, and the writing was clear and helpful! This was a delightful guide to work with.

You'll find my suggestions below, most of them quite small. GitHub is also not the *greatest* way to edit some of this, especially formatting. I know for example that code samples inside numbered lists will all need to be indented a couple of times... but let's save that until we've got a polished product, then take care of the *really* small things like that!  😅 

Thank you again for this wonderful contribution, and I know it might take some time to work through my review comments. No rush on my end! Whenever you get around to it is great!

**Review by mlingham79 — COMMENTED** [review]
@sarah11918 , Thanks very much for your review and great suggestions. I've committed them and made a small edit to include the Astro docs link for dynamic routing.  I'm looking forward to the approval of these changes.

**Comment by sarah11918** [comment]
This looks great, @mlingham79! Just needed to reread and check indentation levels of the code samples, but I think it's in great shape now. We are very happy to officially welcome you to Team Docs!  🥳

**Pull request merged**